mindsdb/interfaces/knowledge_base/preprocessing/document_preprocessor.py

```diff
@@ -119,23 +142,36 @@ def __init__(self, config: ContextualConfig):
         """Initialize with contextual configuration"""
         super().__init__()
         self.config = config
-        self.splitter = FileSplitter(FileSplitterConfig(
-            chunk_size=config.chunk_size,
-            chunk_overlap=config.chunk_overlap
-        ))
-        self.llm = create_chat_model({
-            "model_name": self.config.llm_config.model_name,
-            "provider": self.config.llm_config.provider,
-            **self.config.llm_config.params
-        })
+        self.splitter = FileSplitter(
+            FileSplitterConfig(
+                chunk_size=config.chunk_size, chunk_overlap=config.chunk_overlap
+            )
+        )
+        OpenAI()
+        self.llm = create_chat_model(
+            {
+                "model_name": self.config.llm_config.model_name,
+                "provider": self.config.llm_config.provider,
+                **self.config.llm_config.params,
+            }
+        )
         self.context_template = config.context_template or self.DEFAULT_CONTEXT_TEMPLATE
         self.summarize = self.config.summarize
 
-    def _generate_context(self, chunk_content: str, full_document: str) -> str:
+    def _generate_context(
+        self, chunk_contents: list[str], full_documents: list[str]
+    ) -> list[str]:
         """Generate contextual description for a chunk using LLM"""
-        prompt = self.context_template.replace("{{WHOLE_DOCUMENT}}", full_document)
-        prompt = prompt.replace("{{CHUNK_CONTENT}}", chunk_content)
-        response = self.llm(prompt)
+        prompts = [
```

Comment:
AttributeError: 'list' object has no attribute 'replace' 

You can run [tests/unit/test_document_preprocessor.py](https://github.com/mindsdb/mindsdb/blob/ed7cdb2fa3ff715a1e8fa3fab649418e937929a7/tests/unit/test_document_preprocessor.py) to validate that changes haven't broken anything